src/ctlib/unittests/cs_convert.c

```diff
@@ -296,11 +296,13 @@ main(void)
 	DO_TEST(CS_INT test = 1234567;
 		CS_CHAR test2[] = "1234567", CS_INT_TYPE, &test, sizeof(test), CS_CHAR_TYPE, 7, CS_SUCCEED, test2, 7);
 	DO_TEST(CS_CHAR test[] = "abc";
-		CS_CHAR test2[] = "ab", CS_CHAR_TYPE, test, 3, CS_CHAR_TYPE, 2, CS_FAIL, test2, 2);
+		CS_CHAR test2[] = "ab", CS_CHAR_TYPE, test, 3, CS_CHAR_TYPE, 2,
+		CS_SUCCEED /* CS_FAIL */, test2, 2);
```

Comment:
I think these tests are checked against proprietary libraries. That is you are introducing a bug.

Reply:
Ah, yeah, looks like such truncation should occur only in the context of bulk insertion (where it *does* happen with proprietary libraries).